qiita_ware/processing_pipeline.py

```diff
@@ -158,13 +184,70 @@ def _get_preprocess_fastq_cmd(raw_data, prep_template, params):
     forward_seqs = sorted(forward_seqs)
     reverse_seqs = sorted(reverse_seqs)
     barcode_fps = sorted(barcode_fps)
-    mapping_fps = sorted(mapping_fps)
 
-    # Create the split_libraries_fastq.py command
-    cmd = str("split_libraries_fastq.py --store_demultiplexed_fastq -i %s -b "
-              "%s -m %s -o %s %s"
-              % (','.join(forward_seqs), ','.join(barcode_fps),
-                 ','.join(mapping_fps), output_dir, params_str))
+    # Create a temporary directory to store the split libraries output
+    output_dir = mkdtemp(
+        dir=qiita_config.working_dir,
+        prefix='slq_out_%d_%d_' % (prep_template.id, raw_data.id))
+
+    # Add any other parameter needed to split libraries fastq
+    params_str = params.to_str()
+
+    if len(forward_seqs) == 0:
+        raise ValueError("Forward reads file not found on raw data %s"
+                         % raw_data.id)
+
+    if filetype == "per_sample_FASTQ":
+        if barcode_fps:
+            raise ValueError("per_sample_FASTQ can not have barcodes:"
+                             % (', '.join(barcode_fps)))
+
+        sn_by_rp = _get_sample_names_by_run_prefix(prep_template)
+        samples = []
+        for f in forward_seqs:
+            # getting just the main filename
+            f = basename(f).split('_', 1)[1]
+            # removing extentions: fastq or fastq.gz
+            if 'fastq' in f.rsplit('.', 2):
+                f = f[:f.rindex('.fastq')]
+            # this try/except block is simply to retrieve all possible errors
+            # and display them in the next if block
+            try:
+                samples.append(sn_by_rp[f])
+                del sn_by_rp[f]
+            except KeyError:
+                pass
+
+        if sn_by_rp:
+            raise ValueError(
+                'Some run_prefix values do not match your sample names: %s'
```

Comment:
The code above is not really looking for a "prefix", it's looking for an exact match of the file name once the extensions have been removed. I think the code should be changed to search by prefix and not by exact match. Because a file like `sample1_L001_R1.fastq.gz` would not be found above even though the run_prefix was set as `sample1`.

An alternative, is that we may be using the term `run_prefix` incorrectly here as we are not exactly looking for a prefix.